lucene/core/src/java/org/apache/lucene/document/LatLonPointQuery.java

```
@@ -0,0 +1,174 @@
+    NumericUtils.intToSortableBytes(encodeLongitude(component2D.getMinX()), minLon, 0);
+    NumericUtils.intToSortableBytes(encodeLongitude(component2D.getMaxX()), maxLon, 0);
+    
+    return new SpatialVisitor() {
+      @Override
+      protected Relation relate(byte[] minPackedValue, byte[] maxPackedValue) {
+        if (Arrays.compareUnsigned(minPackedValue, 0, Integer.BYTES, maxLat, 0, Integer.BYTES) > 0 ||
+                Arrays.compareUnsigned(maxPackedValue, 0, Integer.BYTES, minLat, 0, Integer.BYTES) < 0 ||
+                Arrays.compareUnsigned(minPackedValue, Integer.BYTES, Integer.BYTES + Integer.BYTES, maxLon, 0, Integer.BYTES) > 0 ||
+                Arrays.compareUnsigned(maxPackedValue, Integer.BYTES, Integer.BYTES + Integer.BYTES, minLon, 0, Integer.BYTES) < 0) {
+          // outside of global bounding box range
+          return Relation.CELL_OUTSIDE_QUERY;
+        }
+
+        double cellMinLat = decodeLatitude(minPackedValue, 0);
+        double cellMinLon = decodeLongitude(minPackedValue, Integer.BYTES);
+        double cellMaxLat = decodeLatitude(maxPackedValue, 0);
+        double cellMaxLon = decodeLongitude(maxPackedValue, Integer.BYTES);
+
+        return component2D.relate(cellMinLon, cellMaxLon, cellMinLat, cellMaxLat);
+      }
+
+      @Override
+      protected Predicate<byte[]> intersects() {
+        return packedValue -> component2DPredicate.test(NumericUtils.sortableBytesToInt(packedValue, 0),
+                NumericUtils.sortableBytesToInt(packedValue, Integer.BYTES));
+      }
+
+      @Override
+      protected Predicate<byte[]> within() {
+        return packedValue -> component2DPredicate.test(NumericUtils.sortableBytesToInt(packedValue, 0),
+                NumericUtils.sortableBytesToInt(packedValue, Integer.BYTES));
+      }
+
+      @Override
+      protected Function<byte[], Component2D.WithinRelation> contains() {
+        return packedValue -> component2D.withinPoint(
```

It doesn't make sense for ```LatLonPointQuery``` to search for Points that contain any geometry other than points. I'd throw an exception if the incoming geometry type is not a point or array of points.

---

Make sense and makes implementation cleaner. Still I return a MatchNoDocs query for the Query builders. Change is in 3e97bb8.

---

This looks good!